### PR TITLE
Fix SpGEMM crashing on empty rows

### PR DESCRIPTION
**Pull request by MikolajZuzek**
## Contents

* SpGEMM Unit test:
  - [x] include `SPGEMM_KK_LP` to cover `MultiCoreTag4` functor in hash method on CPU;
  - [x] include `SPGEMM_KK_SPEED` to cover speed method - `KokkosSPGEMM_numeric_speed()`;
  - [x] Add case with 0-nnz non-empty arrays (empty rows);
* SpGEMM fixes:
  - [x] skip spgemm numeric phase altogether if any matrix has zero rows or columns;
  - [x] skip computation for empty rows in `GPUTag4` (`SPGEMM_KK_MEMORY_SPREADTEAM`) and `GPUTag6` (`SPGEMM_KK_MEMORY_BIGSPREADTEAM`)
  - [x] align shared memory size according to Kokkos requirements.

### Shared memory alignment

`teamMember.team_shmem().get_shmem(shared_memory_size)` aligns the size to 8 bytes rounding up and if this number is higher than allocation size in `team_shmem_size()`, the allocation will fail returning `nullptr` (and crash follow-up computation). Thus we need to always align our `shared_memory_size`.
> Is this semantic and requirement [documented in Kokkos](https://github.com/kokkos/kokkos/wiki/HierarchicalParallelism#83-team-scratch-pad-memory) clearly enough ?

### Empty row guard

Without proposed guards that skip empty rows, `GPUTag4` and `GPUTag6` can crash trying to fetch `entriesA`/`valuesA` as `a_col_begin_offset` in case of empty row indexes the first element _of the next row_ and - when A has trailing empty rows - *falls outside the view bounds*:
```cpp
      const size_type a_col_begin_offset = row_mapA[row_index];

      nnz_lno_t a_col_ind = entriesA[a_col_begin_offset]; // << index out of bounds
      scalar_t a_col_val  = valuesA[a_col_begin_offset];  // << when A has trailing empty row(s)
```

### CODE REVIEW

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

File: src/sparse/KokkosSparse_spgemm_numeric.hpp [hunk 1]
```diff
@@ -139,6 +139,8 @@ void spgemm_numeric(KernelHandle *handle,
         "If you need this case please let kokkos-kernels developers know.\n");
   }
 
+  if (m < 1 || n < 1 || k < 1) return;
```
Comment (lucbv) [inline src/sparse/KokkosSparse_spgemm_numeric.hpp:142]: I do not recall the exact interface we provide so I just want to make sure that the short cut above does not create issues for something like: `C=beta*C+alpha*A*B` which would fail to compute `C=beta*C` if `k==0`.
I believe that we only implement `C=alpha*A*B` though?

**Review by lucbv — COMMENTED** [review]
This looks good I just want to make sure we are not missing a corner case, also @brian-kelley please look at this too.

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED by label AT: PRE-TEST INSPECTED! Autotester is Removing Label; This inspection will remain valid until a new commit to source branch is performed.

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_CUDA9_GCC720_Light_Tpls_GCC720_GCC740</h3>
<ul>
  <li>Build Num: 16</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_CUDA10_Tpls_CUDA10_LayoutRight</h3>
<ul>
  <li>Build Num: 16</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GCC720</h3>
<ul>
  <li>Build Num: 674</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GCC720_Light_LayoutRight</h3>
<ul>
  <li>Build Num: 321</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_GCC720</h3>
<ul>
  <li>Build Num: 665</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_INTEL18</h3>
<ul>
  <li>Build Num: 652</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CLANG1001</h3>
<ul>
  <li>Build Num: 58</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: KOKKOSKERNELS (NexGenAnalytics/kokkos-kernels)</dt>
  <ul>
    <li>Branch: fix-spgemm-empty-row-crash</li>
    <li>SHA: 7fcea130bd81bdb68afce92b1ff483043f9e0bf6</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: MikolajZuzek
</details>

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_CUDA9_GCC720_Light_Tpls_GCC720_GCC740</h3>
<ul>
  <li>Build Num: 16</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_CUDA10_Tpls_CUDA10_LayoutRight</h3>
<ul>
  <li>Build Num: 16</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GCC720</h3>
<ul>
  <li>Build Num: 674</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GCC720_Light_LayoutRight</h3>
<ul>
  <li>Build Num: 321</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_GCC720</h3>
<ul>
  <li>Build Num: 665</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_INTEL18</h3>
<ul>
  <li>Build Num: 652</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CLANG1001</h3>
<ul>
  <li>Build Num: 58</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_BRANCH</td>
    <td>fix-spgemm-empty-row-crash</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/NexGenAnalytics/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>7fcea130bd81bdb68afce92b1ff483043f9e0bf6</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>7413f2a01f1efaa406bfaf85b6e77dba5f41c9b7</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td></td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1220</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>
</details>

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS BEEN REVIEWED, BUT NOT ACCEPTED OR REQUIRES CHANGES

**Comment by kokkos-devops-admin** [comment]
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

File: src/sparse/KokkosSparse_spgemm_numeric.hpp (hunk 1 above)
Comment (brian-kelley) [reply src/sparse/KokkosSparse_spgemm_numeric.hpp:142]: Correct, we don't support adding to an existing C, so this check is OK.

**Review by brian-kelley — APPROVED** [review]
Thanks @MikolajZuzek !

**Review by lucbv — APPROVED** [review]
Thanks for the fix!

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ brian-kelley lucbv ]!

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pull Request AutoTester' - Pull Request MUST BE MERGED MANUALLY BY Project Team - This Repo does not support Automerge
